src/main/java/org/apache/sling/resourceresolver/impl/FactoryRegistrationHandler.java

```diff
@@ -77,15 +117,24 @@ public void close() {
     void maybeRegisterFactory(final String unavailableName, final String unavailableServicePid) {
         if (!factoryRegistrationWorker.isShutdown()) {
             LOG.debug("submitting maybeRegisterFactory");
-            factoryRegistrationWorker.submit(() -> {
-                final boolean preconditionsOk = factoryPreconditions.checkPreconditions(unavailableName, unavailableServicePid);
-                if ( preconditionsOk && this.factoryRegistration == null ) {
+            final boolean preconditionsOk;
+            final ResourceResolverFactoryActivator localActivator;
+            try {
+                configurationLock.lock();
+                preconditionsOk = factoryPreconditions.checkPreconditions(unavailableName, unavailableServicePid);
+                localActivator = this.activator;
+            } finally {
+                configurationLock.unlock();
+            }
+
+            factoryRegistrationWorker.execute(() -> {
+                if (preconditionsOk && this.factoryRegistration == null) {
```

Comment:
What are the consequences of making an "unsychronized" read of `this.factoryRegistration` in this method?

Reply:
Again, same as abov. I'll add the `registrationLock` here.